src/ahp/prover.rs

```diff
@@ -62,14 +62,35 @@ pub enum ProverMsg<F: Field> {
 }
 
 impl<F: Field> algebra::ToBytes for ProverMsg<F> {
-    fn write<W: std::io::Write>(&self, w: W) -> std::io::Result<()> {
+    fn write<W: Write>(&self, mut writer: W) -> std::io::Result<()> {
         match self {
-            ProverMsg::EmptyMessage => Ok(()),
-            ProverMsg::FieldElements(field_elems) => field_elems.write(w),
+            ProverMsg::EmptyMessage => 0u32.write(&mut writer),
```

Comment:
We write a u32 zero for an empty message? If these four bytes (really four?) encode the length of the message than that's ok.